packages/microservices/server/server-nats.ts

```diff
@@ -21,29 +17,37 @@ import { Server } from './server';
 
 let natsPackage = {} as any;
 
+// To enable type safety for Nats. This cant be uncommented by default
+// because it would require the user to install the nats package even if they dont use Nats
+// Otherwise, TypeScript would fail to compile the code.
+//
+// type Client = import('nats').NatsConnection;
+// type NatsMsg = import('nats').Msg;
+type Subscription = import('nats').Subscription;
+
+type Client = any;
+type NatsMsg = any;
+// type Subscription = any;
+
 /**
  * @publicApi
  */
-export class ServerNats extends Server implements CustomTransportStrategy {
+export class ServerNats extends Server<NatsEvents, NatsStatus> {
```

Comment:
```suggestion
export class ServerNats<E = NatsEvents, S = NatsStatus> extends Server<E, S> {
```

Minor nitpick, but JetStream has additional events and adding the generic would make it slightly easier to forward [ConsumerEvents](https://github.com/nats-io/nats.deno/blob/main/jetstream/consumer.ts#L182) along the same tracks.

Reply:
I was pondering the same thing, but I don't think other transports have a superset of functionality like JetStream.

Adding the generic would then be for a developer to add additional, custom events to the Server class. I can imagine adding a health/connection check to your custom Client/Server classes. Or perhaps emitting events from a kind of interceptor you're using to transform the payload/result. 

It's an edge-case for sure, and I wonder how much surface it would expose for future maintainability. I'm sure much of it could also be accomplished by unwrapping the client now that's become possible. 

The same goes here, which is why it's just a nitpick, and in your situation I'd probably postpone till a later date 😄 it's always hard to foresee 2nd and 3rd level effects.

Reply:
Make sense, thanks Rick!